### ISSUE TITLE
a11y: va-toast

### ISSUE THREAD

**Issue by kushich** [epicmaxco/vuestic-ui#3287]
- [x] step 1: investigate a11y guides for this component;
- [x] step 2: discuss with team;
- [ ] step 3: implement required functionality;
- [ ] step 4: add docs page section about a11y.

**Comment by kushich** [comment]
[ref](https://getbootstrap.com/docs/4.3/components/toasts/#accessibility)
> Toasts are intended to be small interruptions to your visitors or users, so to help those with screen readers and similar assistive technologies, you should wrap your toasts in an [aria-live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions). Changes to live regions (such as injecting/updating a toast component) are automatically announced by screen readers without needing to move the user’s focus or otherwise interrupt the user. Additionally, include `aria-atomic="true"` to ensure that the entire toast is always announced as a single (atomic) unit, rather than announcing what was changed (which could lead to problems if you only update part of the toast’s content, or if displaying the same toast content at a later point in time).

> Note that the live region needs to be present in the markup before the toast is generated or updated. If you dynamically generate both at the same time and inject them into the page, they will generally not be announced by assistive technologies.

> You also need to adapt the role and aria-live level depending on the content. If it’s an important message like an error, use `role="alert"` `aria-live="assertive"`, otherwise use `role="status"` `aria-live="polite"` attributes (_In our case should be configurable by props_).

And for the accessibility section in docs we should mention that it's better to use `VaAlert` which provides the same function as the `VaToast` component, but is not timed which allows users an adequate amount of time to view the message without it being timed-out (or `VaToast` without timeout).

**Comment by asvae** [comment]
> And for the accessibility section in docs we should mention that it's generally a better idea to use VaAlert which provides the same function as the VaToast component, but is not timed which allows users an adequate amount of time to view the message without it being timed-out (or VaToast without timeout).

Please check that with refs. I'm under impression that toast and alert solve different problems, so it's hard to tell "prefer one vs another".

**Comment by xiongmao86** [comment]
Hi, @asvae, I would like to take a stab of this, but I am not sure how to handle the `role` of `VaToast`, as @kushich mention the role would be `alert` or `status`, but the role in code could be `alertdialog` and `alert`, depending on `$props.closeable`.

 I am not sure how to reconcile this gap, should I add a `$prop.important` to set `:role="$props.closable ? 'alertgialog' : $props.important ? 'alert' : 'status'"`. Or as mentioned in docs of bootstrap just use 'status' and relies `VaAlert` to show important message.

**Comment by m0ksem** [comment]
> Hi, @asvae, I would like to take a stab of this, but I am not sure how to handle the `role` of `VaToast`, as @kushich mention the role would be `alert` or `status`, but the role in code could be `alertdialog` and `alert`, depending on `$props.closeable`.
> 
> I am not sure how to reconcile this gap, should I add a `$prop.important` to set `:role="$props.closable ? 'alertgialog' : $props.important ? 'alert' : 'status'"`. Or as mentioned in docs of bootstrap just use `status` and relies `VaAlert` to show important message. Or are there any other way to handle this situation?

I think role must be determinate automatically based on props like `closable`, but user still able to change it any moment with attributes. So you don't need to handle “importance”, better give ability to set `role` manually.

```vue
<VaToast role="status" />
```

**Comment by xiongmao86** [comment]
Thanks @m0ksem, I have a pr here.
